src/main/resources/data/computercraft/lua/rom/programs/fun/speaker.lua

```diff
@@ -40,12 +48,60 @@ elseif cmd == "play" then
         error(err, 0)
     end
 
+    local start = handle.read(4)
+    local pcm = false
+    if start == "RIFF" then
+        handle.read(4)
+        if handle.read(8) ~= "WAVEfmt " then
+            handle.close()
+            error("Could not play audio: Unsupported WAV file", 0)
+        end
+
+        local fmtsize = ("<I4"):unpack(handle.read(4))
+        local fmt = handle.read(fmtsize)
+        local format, channels, rate, _, _, bits = ("<I2I2I4I4I2I2"):unpack(fmt)
+        if not ((format == 1 and bits == 8) or (format == 0xFFFE and bits == 1)) then
+            handle.close()
+            error("Could not play audio: Unsupported WAV file", 0)
+        end
+        if channels ~= 1 or rate ~= 48000 then
+            print("Warning: This WAV file may not play correctly.")
```

Comment:
Maybe there should be a little more detail in this message? Something like "Only WAV files with 1 channel sampled at 48kHz are supported. This file may play incorrectly". 

Not sure, it's a little verbose!

Reply:
I shortened it a little bit, but hopefully people can look up what it means (or ask for help - preferably not in your DMs though!).